engine/consensus/ingestion/core_test.go

```diff
@@ -210,29 +214,6 @@ func (suite *IngestionCoreSuite) TestOnGuaranteeNoGuarantors() {
 	suite.pool.AssertNotCalled(suite.T(), "Add", guarantee)
 }
 
-// TestOnGuaranteeInvalidRole verifies that a collection is rejected if any of
-// the signers has a role _different_ than collection.
-// We expect an engine.InvalidInputError.
-func (suite *IngestionCoreSuite) TestOnGuaranteeInvalidRole() {
```

Comment:
Those cases are tested by unit-test thoroughly already. 

Case 1 is tested by [This](https://github.com/onflow/flow-go/pull/2414/files#diff-f70d80e912ec8de76d56a70b94a4d33e2d4b9615137e131f26988b281253ff36R206)
Case 2 is tested by [This](https://github.com/onflow/flow-go/pull/2414/files#diff-f70d80e912ec8de76d56a70b94a4d33e2d4b9615137e131f26988b281253ff36R151)
Case 3 is tested by [This](https://github.com/onflow/flow-go/pull/2414/files#diff-3053425efedee6c35a204fadecebb6173eaa5254d2e585b68c9d46eb83457008R67)
Case 4 is not possible for now, because `DecodeSignerIndicesToIdentities` don't return any error other than the above.

Reply:
Maybe I am looking at the wrong thing, but for me, the links don't lead to meaningful sections of the code. Could you directly reference the code pieces, please? thanks

Reply:
The purpose of these tests that I am suggesting is _not_ to verify that `FindGuarantors` returns the correct sentinel error. The purpose is to test that `ingestion.Core` correctly handles the errors returned by `FindGuarantors`:
   * for _expected_ sentinel errors, `ingestion.Core` wraps them as `InvalidInputError ` 
   * `ingestion.Core` does _not_ misinterpret unexpected exceptions as the input being invalid

Reply:
> It would be good to test correct handling of `4 unexpected exception`

If what you meant is that: we should test if `DecodeSignerIndicesToIdentities` returns an unexpected exception, then ingestione.Core's `OnGuarantee` should also return an exception rather than a `InvalidInputError` error. 

My point was that this test case is not possible to make, because in order to create it, we need the current `DecodeSignerIndicesToIdentities` implementation to return an exception, which it doesn't. In other words, it doesn't return any error, other than `ErrIllegallyPaddedBitVector`, `ErrIncompatibleBitVectorLength` and `ErrInvalidChecksum`. 

We also can't mock `DecodeSignerIndicesToIdentities` to return exception, because it's a pure function.

Reply:
For case 1 to 3, I only created one of them which let `DecodeSignerIndicesToIdentities` to return `ErrInvalidChecksum` error, and checked that the returned error is a `ErrInvalidChecksum` wrapped as `InvalidInputError`. 

I don't think we need to test all 3 of them, but just one. Because all three of them will be checked by the `IsDecodeSignerIndicesError` function, and handled the same way. And I have tests testing the 3 cases in the unittest of the `DecodeSignerIndicesToIdentities` itself. Testing all 3 of them here would be over-testing. I think it's ideal to maintain minimum tests, and maximize their coverage.